src/oci/oci_runtime.rs

```diff
@@ -0,0 +1,675 @@
+    Detach,
+    PidFile(PathBuf),
+    NoSubreaper,
+    NoPivot,
+    NoNewKeyright,
+    PreserveFds(u32),
+}
+
+impl fmt::Display for RunArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::oci_runtime::RunArgs::*;
+        write!(f, "--")?;
+
+        match self {
+            Bundle(path) => write_kv(f, self, path.display()),
+            ConsoleSocket(path) => write_kv(f, self, path.display()),
+            PidFile(path) => write_kv(f, self, path.display()),
+            PreserveFds(val) => write_kv(f, self, val),
+            _ => write!(f, "{}", self.as_ref()),
+        }
+    }
+}
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+/// Available arguments for 'oci_runtime spec'.
+pub enum SpecArgs {
+    Bundle(PathBuf),
+    Rootless,
+}
+
+impl fmt::Display for SpecArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::oci_runtime::SpecArgs::*;
+        write!(f, "--")?;
+
+        match self {
+            Bundle(path) => write_kv(f, self, path.display()),
+            _ => write!(f, "{}", self.as_ref()),
+        }
+    }
+}
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+/// Available arguments for 'oci_runtime update'.
+pub enum UpdateArgs {
+    BlkioWeight(u16),
+    CpuPeriod(u64),
+    CpuQuota(u64),
+    CpuShare(u64),
+    CpuRtPeriod(u64),
+    CpuRtRuntime(u64),
+    CpusetCpus(u64),
+    CpusetMems(u64),
+    KernelMemory(u64),
+    KernelMemoryTcp(u64),
+    Memory(u64),
+    MemoryReservation(u64),
+    MemorySwap(u64),
+    PidsLimit(u32),
+    L3CacheSchema(String),
+    MemBwSchema(String),
+}
+
+impl fmt::Display for UpdateArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::oci_runtime::UpdateArgs::*;
+        write!(f, "--")?;
+
+        match self {
+            BlkioWeight(val) => write_kv(f, self, val),
+            CpuPeriod(val) => write_kv(f, self, val),
+            CpuQuota(val) => write_kv(f, self, val),
+            CpuShare(val) => write_kv(f, self, val),
+            CpuRtPeriod(val) => write_kv(f, self, val),
+            CpuRtRuntime(val) => write_kv(f, self, val),
+            CpusetCpus(val) => write_kv(f, self, val),
+            CpusetMems(val) => write_kv(f, self, val),
+            KernelMemory(val) => write_kv(f, self, val),
+            KernelMemoryTcp(val) => write_kv(f, self, val),
+            Memory(val) => write_kv(f, self, val),
+            MemoryReservation(val) => write_kv(f, self, val),
+            MemorySwap(val) => write_kv(f, self, val),
+            PidsLimit(val) => write_kv(f, self, val),
+            L3CacheSchema(val) => write_kv(f, self, val),
+            MemBwSchema(val) => write_kv(f, self, val),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    //TODO
+    use super::*;
+    use std::{os::unix::process::ExitStatusExt, process::ExitStatus};
```

Comment:
yeah. I need more test. I agree. suggestions here? only so much I can test here.

Reply:
I think testing some main logical paths would be fine by mocking the exec.